ecommerce/extensions/payment/views/paytabs.py

```diff
@@ -0,0 +1,125 @@
+"""PayTabs response processing views."""
+import logging
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
+from django.urls import reverse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from oscar.apps.partner import strategy
+from oscar.core.loading import get_class, get_model
+from pypaytabs import Paytabs
+
+from ecommerce.extensions.checkout.mixins import EdxOrderPlacementMixin
+from ecommerce.extensions.checkout.utils import get_receipt_page_url
+from ecommerce.extensions.payment.processors.paytabs import PayTabs
+
+logger = logging.getLogger(__name__)
+
+Applicator = get_class('offer.applicator', 'Applicator')
+Basket = get_model('basket', 'Basket')
+OrderNumberGenerator = get_class('order.utils', 'OrderNumberGenerator')
+OrderTotalCalculator = get_class('checkout.calculators', 'OrderTotalCalculator')
+NoShippingRequired = get_class('shipping.methods', 'NoShippingRequired')
+
+
+class PayTabsResponseView(EdxOrderPlacementMixin, View):
+    """
+    View to handle the response from PayTabs after processing the payment.
+    """
+    @property
+    def payment_processor(self):
+        return PayTabs(self.request.site)
+
+    @method_decorator(transaction.non_atomic_requests)
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super(PayTabsResponseView, self).dispatch(request, *args, **kwargs)
+
+    def _verify_response(self, request, payment_reference):
+        """
+        Verify the given payment_reference number to confirm that it is for a valid transaction
+        and return the verification response data.
+        """
+        partner_short_code = request.site.siteconfiguration.partner.short_code
+        configuration = settings.PAYMENT_PROCESSOR_CONFIG[partner_short_code.lower()][self.payment_processor.NAME]
+        paytabs_client = Paytabs(configuration['merchant_email'], configuration['secret_key'])
```

Comment:
if we could store `merchant_email` and `secret_key` in variable and then pass it to the `Paytabs` class, it will be cleaner.